src/completion.ts

```diff
@@ -63,7 +63,7 @@ export function asCompletionItem(entry: tsp.CompletionEntry, optionalReplacement
     }
 
     let insertText = entry.insertText;
-    const replacementSpan = entry.replacementSpan || optionalReplacementSpan;
+    const replacementSpan = entry.replacementSpan || (features.completionInsertReplaceSupport ? optionalReplacementSpan : undefined);
```

Comment:
I would suggest removing unnecessary code bellow:

```diff
- const replacementSpan = entry.replacementSpan || (features.completionInsertReplaceSupport ? optionalReplacementSpan : undefined);
+ const replacementSpan = entry.replacementSpan || optionalReplacementSpan;

...

- if (replacementRange) {
+ if (replacementRange && features.completionInsertReplaceSupport) {
    if (!insertText) {
        insertText = item.label;
    }
-    if (features.completionInsertReplaceSupport) {
      const insertRange = lsp.Range.create(replacementRange.start, position);
      item.textEdit = lsp.InsertReplaceEdit.create(insertText, insertRange, replacementRange);
-    } else {
-        item.textEdit = lsp.TextEdit.replace(replacementRange, insertText);
-    }
} else {
    item.insertText = insertText;
}
```

Reply:
That doesn't sound right. The replacementRange can also come from the `entry.replaceSpan` in which case we want to create a replace TextEdit.

Reply:
True, you are right.

Reply:
Will try out this PR tomorrow.